vertical-pod-autoscaler/pkg/recommender/input/controller_fetcher/controller_fetcher.go

```diff
@@ -82,10 +83,44 @@ type controllerFetcher struct {
 	scaleNamespacer scale.ScalesGetter
 	mapper          apimeta.RESTMapper
 	informersMap    map[wellKnownController]cache.SharedIndexInformer
+	controllerCache controllerCacheStorage
+}
+
+func (f *controllerFetcher) periodicallyRemoveExpiredFromCache(ctx context.Context, period time.Duration) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(period):
+			f.controllerCache.RemoveExpired()
+		}
+	}
+}
+
+func (f *controllerFetcher) periodicallyRefreshCache(ctx context.Context, period time.Duration) {
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(period):
```

Comment:
Our refresh period for an entry is 10 minutes, but we check which entries should be refreshed every 1 hour. Is this intentional?

Reply:
Changed to 1 minute